ppl/zqd/core.go

```diff
@@ -36,11 +36,17 @@ type Config struct {
 	Root        string
 	Version     string
 
+	Auth     AuthConfig
 	Suricata pcapanalyzer.Launcher
 	Zeek     pcapanalyzer.Launcher
 }
 
+type middleware interface {
+	Middleware(next http.Handler) http.Handler
+}
+
 type Core struct {
+	auth      middleware
```

Comment:
I think this could be a little more straightforward if (a) the type of this field were mux.MiddlewareFunc and (b) the field was nil when AuthConfig.Enabled was false. Then we wouldn't need the middleware interface or noopAuthenticator.